includes/admin/meta-boxes/class-wcs-meta-box-related-orders.php

```diff
@@ -115,24 +114,24 @@ public static function output_rows( $post ) {
 			}
 		}
 
-		$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, $post );
+		$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, $order );
```

Comment:
This change is effectively the reason behind the error you mentioned in the PR description and in the corresponding issue (https://github.com/woocommerce/woocommerce-subscriptions/issues/4417):

```
PHP Notice:  Function ID was called <strong>incorrectly</strong>. Order properties should not be accessed directly. Backtrace: require('wp-admin/edit-form-advanced.php'), do_meta_boxes, WCS_Meta_Box_Related_Orders::output, include_once('/plugins/woocommerce-subscriptions-core/includes/admin/meta-boxes/views/html-related-orders-table.php'), do_action('woocommerce_subscriptions_related_orders_meta_box_rows'), WP_Hook->do_action, WP_Hook->apply_filters, WCS_Meta_Box_Related_Orders::output_rows, apply_filters('woocommerce_subscriptions_admin_related_orders_to_display'), WP_Hook->apply_filters, WC_Subscriptions_Switcher::display_switches_in_related_order_metabox, WC_Abstract_Legacy_Order->__get, wc_doing_it_wrong Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 3.0.) in //wp-includes/functions.php on line 5835
```

This is a breaking change and rather than replace the `$post` variable with an order object we will need to deprecate this filter, only call it if there are third-parties hooking onto it and then trigger a new hook where we can pass the order freely. 

This will look like this roughly. 

```php
if ( has_filter( 'woocommerce_subscriptions_admin_related_orders_to_display' ) ) {
	wcs_deprecated_hook( 'woocommerce_subscriptions_admin_related_orders_to_display', 'subscriptions-core 5.0.0', 'wcs_admin_subscription_related_orders_to_display' );

	/**
	 * Filters the orders to display in the Related Orders meta box.
	 *
	 * This filter is deprecated in favour of 'wcs_admin_subscription_related_orders_to_display'.
	 *
	 * @deprecated subscriptions-core 5.0.0
	 *
	 * @param array   $orders_to_display An array of orders to display in the Related Orders meta box.
	 * @param array   $subscriptions An array of subscriptions related to the order.
	 * @param WP_Post $post The order post object.
	 */
	$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, get_post( $order->get_id() ) );
}

/**
 * Filters the orders to display in the Related Orders meta box.
 *
 * @since subscriptions-core 5.0.0
 *
 * @param array    $orders_to_display An array of orders to display in the Related Orders meta box.
 * @param array    $subscriptions An array of subscriptions related to the order.
 * @param WC_Order $order The order object.
 */
$orders_to_display = apply_filters( 'wcs_admin_subscription_related_orders_to_display', $orders_to_display, $subscriptions, $order );
```

Reply:
The `get_post( $order->get_id() )`  will return the order post or the placeholder post on HPOS sites, so on HPOS sites there may be some change in data received, however fully fledged subscription stores running HPOS is far enough down the track that third-parties will likely have replaced uses of this.

This behaviour of post objects becoming unusable is inevitable.

Reply:
Thanks for the guidance here @james-allan, appreciate it!

I've resolved this one using your suggestion above in 76eb615, noting that this may break for HPOS-enabled sites using the deprecated hook.